examples/defaulter-gen/generators/defaulter.go

```diff
@@ -442,16 +442,16 @@ func newCallTreeForType(existingDefaulters, newDefaulters defaulterFuncMap) *cal
 	}
 }
 
-func resolveTypeAndDepth(t *types.Type) (*types.Type, int) {
+func resolveTypeAndDepth(t *types.Type) (*types.Type, []*types.Type) {
 	var prev *types.Type
-	depth := 0
+	var depth []*types.Type
```

Comment:
this is not really depth anymore. What is it?

Reply:
Renamed `defaultDepth` to `pointerElementPath`, and separated this into a separate function `getPointerElementPath` with detailed comments